### ISSUE TITLE
Technical - Material UI React Research

### ISSUE THREAD

**Issue by MichaelFriel** [GroupOneProject/Track-City#5]
We need to research the capabilities of Material UI and establish whether there are some components which we can use straight off the rack.

https://mui.com/store/items/devias-kit/

Can you confirm: 

- Can this be used for free?
- What range of components are there?
- Can we create an app which partially follows this, but still has some bespoke componentry if required?

**Comment by ToussaintPhillips** [comment]
Yes, Material-UI provides a wide range of components that cover various aspects of UI development. Here's a brief overview of the categories of components available in Material-UI:

1. **Layout Components:**
   - Grid: A responsive layout system based on Flexbox.
   - Container: A component to center content horizontally within a viewport.

2. **Navigation Components:**
   - App Bar: A horizontal bar for navigation and other actions.
   - Drawer: A vertical panel displayed on the side of the screen.
   - Tabs: A tabbed navigation interface.

3. **Inputs Components:**
   - Button: A clickable button with various styles and configurations.
   - TextField: An input field for text entry.
   - Checkbox, Radio, Switch: Components for selecting options.
   - Select: A dropdown menu for selecting from a list of options.

4. **Display Components:**
   - Typography: Components for displaying text with different styles and sizes.
   - Icons: A library of material icons for use in your application.
   - Avatar: A graphical representation of a user or entity.
   - Badge: A small status indicator.

5. **Data Display Components:**
   - Table: A data table with sorting, pagination, and other features.
   - Card: A container for organizing content, typically used for displaying information.
   - Tooltip: A small pop-up with additional information.
   - Snackbar: A notification system for displaying messages to the user.

6. **Feedback Components:**
   - Dialog: A modal dialog for user interaction.
   - Snackbar: A notification system for displaying messages to the user.
   - Progress: Components for showing progress, such as progress bars and spinners.

Yes, you can create an app that partially follows Material-UI guidelines while incorporating bespoke components as required. Material-UI provides a flexible and customizable theming system that allows you to adapt its components to match your project's design language. You can override default styles, create custom components, and mix them with Material-UI components seamlessly.

For example, if you have specific design requirements that are not covered by Material-UI's components, you can create custom components using your preferred styling approach (e.g., CSS modules, styled-components) and integrate them into your Material-UI-based app. This way, you can maintain consistency with Material-UI components while still incorporating bespoke elements to meet your project's needs.
